### PR TITLE
JS cleanups

### PR DESCRIPTION
**Pull request by phlax**
- Remove ajax objects from queue on completion

### CODE REVIEW

File: pootle/static/js/shared/utils/fetch.js [hunk 1]
```diff
@@ -29,7 +29,9 @@ function fetch({ url, body, method = 'GET', dataType = 'json', queue = null,
       data: body,
     })
   );
-
+  requests[queueName].done(function(data) {
```
Comment (julen) [inline pootle/static/js/shared/utils/fetch.js]: To conform with ESLint:
```js
requests[queueName].done(() => delete requests[queueName]);
```

Comment (phlax) [inline pootle/static/js/shared/utils/fetch.js]: thanks, ill update

File: pootle/static/js/shared/utils/fetch.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ function fetch({ url, body, method = 'GET', dataType = 'json', queue = null,
       data: body,
     })
   );
-
+  requests[queueName].done(() => delete requests[queueName]);
```
Comment (julen) [inline pootle/static/js/shared/utils/fetch.js]: Actually, [after a read on `delete`](http://bertanguven.com/preventing-memory-leaks-in-javascript-null-vs-delete), it seems like we'd prefer to nullify the value `requests[queueName] = null`.

**Comment by phlax** [comment]
contributed to resolving #4916

**Pull request merged**